backend/src/apiserver/client_manager.go

```diff
@@ -119,6 +120,11 @@ func (c *ClientManager) UUID() util.UUIDGeneratorInterface {
 	return c.uuid
 }
 
+func (c *ClientManager) RandomString() util.RandomStringInterface{
```

Comment:
I strongly recommend getting rid of this ClientManager abstraction. It adds unnecessary complexity to the code. This function is a perfect example of code indirection that adds to cognitive burden for the reader of the code. There is no need for generating a random string to be a function of ClientManager. We can push this to its own function where it is used, and in tests, replace that with a mock as needed, i.e.

file where used:
```
var randomStringGenFn = func() { return random() }
```

and then in tests:
```
randomStringGenFn = func() { return "abc"}
```

Reply:
I would +100 for getting rid of it. maybe do it separately?